### PR TITLE
Feature/string or

### PR DESCRIPTION
**Pull request by danieleguido**
Add OR feature for string filters, ready to be integrated as soon as https://github.com/impresso/impresso-middle-layer/pull/251 is integrated into impresso-middle-layer.

### CODE REVIEW

File: src/components/modules/FilterMonitor.vue [hunk 1]
```diff
@@ -240,10 +278,35 @@ export default {
           items: selectedItems,
           q: selectedItemsIds
         })
+      } else if (StringTypes.includes(type)) {
+        const newFilter = {
+          ...this.editedFilter,
+          q: this.editedFilter.q.concat(this.validStringsToAdd.map(d => d.uid)),
+        };
+        console.info('applyChanges', newFilter);
```
Comment (theorm) [inline src/components/modules/FilterMonitor.vue]: Maybe remove the `console.log`?

Comment (danieleguido) [reply src/components/modules/FilterMonitor.vue]: oups...

File: src/models/FilterString.js [hunk 2]
```diff
@@ -4,22 +4,39 @@
...
     super(args);
-    this.q = String(args.q || args.query);
     this.precision = precisions[(args.precision || 'exact').toUpperCase()];
     this.distance = Math.max(0, Math.min(parseInt(args.distance || 0, 10), 10));
   }
 
+  setItems(items = []) {
+    this.items = items.map((uid) => {
+      const item = new StringItem({ uid });
+      item.checked = true;
```
Comment (theorm) [inline src/models/FilterString.js]: What is the `checked` flag for? Looks like it is always true. Do we need it?

Comment (danieleguido) [reply src/models/FilterString.js]: true, I'll remove this, it was related to the previous implementation, using checkboxes

Comment (theorm) [reply src/models/FilterString.js]: Yeah it has been redone with lists which makes it a bit cleaner.

File: src/models/FilterString.js [hunk 3]
```diff
@@ -11,8 +11,10 @@ import * as contexts from './Contexts';
 class StringItem {
   constructor({
     uid = '',
+    checked = true,
   }) {
     this.uid = uid;
+    this.checked = Boolean(checked);
```
Comment (theorm) [inline src/models/FilterString.js]: Is it used?

Comment (danieleguido) [reply src/models/FilterString.js]: Yes, in FilterMonitor finally, to check/unckeck items temporarily:

Screenshot 2020-03-16 at 09 25 59

Comment (theorm) [reply src/models/FilterString.js]: Right. It would be great to move this temporary state into the component at some point.

Comment (danieleguido) [reply src/models/FilterString.js]: yep I saw you started doing that, listing all the `excludedIds`

**Pull request merged**